pnc-rest/src/main/java/org/jboss/pnc/rest/endpoint/UserEndpoint.java

```diff
@@ -49,6 +53,7 @@ public UserRest getSpecific(@ApiParam(value = "User id", required = true) @PathP
     @ApiOperation(value = "Creates new User")
     @POST
     public Response createNew(@NotNull @Valid UserRest userRest, @Context UriInfo uriInfo) {
+        validateIdIsNull(userRest);
```

Comment:
Just a thought - wouldn't be better to validate inside `userProvider`? This way the endpoint would be anemic and testing might be done one step lower - on provider layer. We wouldn't need Arquillian and possibly plain JUnit would be sufficient this way...